util2.py

```diff
@@ -0,0 +1,17 @@
+def parseBoolValue(value: str | None, fail_on_errors: bool = True, preserve_none: bool = False) -> bool | None:
+    """Parses a string representing a boolean value. If parsing was successful,
+       returns `True` or `False`. If `preserve_none=True`, returns `True`, `False`,
+       or `None`. If parsing was not successful, raises `ValueError`, or, if
+       `fail_on_errors=False`, returns `None`."""
+    if not isinstance(value, str):
+        if preserve_none and value is None:
+            return value
+        return bool(value)
+    elif preserve_none and value.lower() == 'none':
+        return None
+    elif value.lower() in ('true', 'yes', 'y', 'on', '1'):
+        return True
+    elif value.lower() in ('false', 'no', 'n', 'off', '0', 'none'):
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/rkj6124/test/pull/2#issuecomment-2421462389>#102274</a><span> - 10/18/2024, 05:55 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The function treats 'none' as a falsy value, which contradicts the behavior when preserve_none is True and may lead to unexpected results.
</div>
<div id="fix">
<b>Fix:</b> Remove 'none' from the falsy values list and handle it separately to maintain consistency with the preserve_none behavior.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -14,6 +14,8 @@
 -    elif value.lower() in ('false', 'no', 'n', 'off', '0', 'none'):
 +    elif value.lower() in ('false', 'no', 'n', 'off', '0'):
      return False
 +    elif value.lower() == 'none' and not preserve_none:
 +        return False
  elif fail_on_errors:
      raise ValueError('Cannot parse bool value: %r' % value)
 +    return None
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged